plugins/modules/autoscaling_instance_refresh.py

```diff
@@ -84,123 +98,143 @@
+            sample: "Pending"
+        preferences:
+            description: The preferences for an instance refresh.
+            returned: always
+            type: dict
+            sample: {
+                'AlarmSpecification': {
+                    'Alarms': [
+                        'my-alarm',
+                    ],
+                },
+                'AutoRollback': True,
+                'InstanceWarmup': 200,
+                'MinHealthyPercentage': 90,
+                'ScaleInProtectedInstances': 'Ignore',
+                'SkipMatching': False,
+                'StandbyInstances': 'Ignore',
+            }
+        start_time:
+            description: The date and time this ASG was created, in ISO 8601 format.
+            returned: success
+            type: str
+            sample: "2015-11-25T00:05:36.309Z"
+        end_time:
+            description: The date and time this ASG was created, in ISO 8601 format.
+            returned: success
+            type: str
+            sample: "2015-11-25T00:05:36.309Z"
+        percentage_complete:
+            description: the % of completeness
+            returned: success
+            type: int
+            sample: 100
+        instances_to_update:
+            description: num. of instance to update
+            returned: success
+            type: int
+            sample: 5
 """
 
-try:
-    from botocore.exceptions import BotoCoreError
-    from botocore.exceptions import ClientError
-except ImportError:
-    pass  # caught by AnsibleAWSModule
+from typing import Dict
+from typing import Optional
+from typing import Union
 
 from ansible.module_utils.common.dict_transformations import camel_dict_to_snake_dict
 from ansible.module_utils.common.dict_transformations import snake_dict_to_camel_dict
 
-from ansible_collections.amazon.aws.plugins.module_utils.retries import AWSRetry
+from ansible_collections.amazon.aws.plugins.module_utils.autoscaling import AnsibleAutoScalingError
+from ansible_collections.amazon.aws.plugins.module_utils.autoscaling import cancel_instance_refresh
+from ansible_collections.amazon.aws.plugins.module_utils.autoscaling import describe_instance_refreshes
+from ansible_collections.amazon.aws.plugins.module_utils.autoscaling import start_instance_refresh
 from ansible_collections.amazon.aws.plugins.module_utils.transformation import scrub_none_parameters
 
 from ansible_collections.community.aws.plugins.module_utils.modules import AnsibleCommunityAWSModule as AnsibleAWSModule
 
 
-def start_or_cancel_instance_refresh(conn, module):
+def validate_healthy_percentage(preferences: Dict[str, Union[bool, int]]) -> Optional[str]:
+    min_healthy_percentage = preferences.get("min_healthy_percentage")
+    max_healthy_percentage = preferences.get("max_healthy_percentage")
+
+    if min_healthy_percentage is not None and (min_healthy_percentage < 0 or min_healthy_percentage > 90):
+        return "The value range for the min_healthy_percentage is 0 to 90."
```

Comment:
Should the min_healthy_percentage value range between 0 and 90 or 0 and 100? I understand the default value is 90. However, what is the maximum allowable value for this parameter?